rust/main/agents/relayer/src/msg/op_submitter.rs

```diff
@@ -711,8 +715,16 @@ async fn confirm_lander_task(
         let confirmed_operations = Arc::new(Mutex::new(0));
         let confirm_futures = payload_status_results
             .into_iter()
-            .map(|(op, status_result)| async {
-                let Ok(payload_status) = status_result else {
+            .map(|(op, status_results)| async {
+                let (successes, failures): (Vec<_>, Vec<()>) =
+                    status_results.into_iter().partition_map(|status| {
+                        if let Ok(payload_status) = status {
+                            return Either::Left(payload_status);
+                        }
+                        Either::Right(())
+                    });
+
+                if !failures.is_empty() {
```

Comment:
- since we don't actually use `failures` for anything other than their count, we could take the initial length of `status_results` and use `filter` for `Ok` responses instead of `partition_map`. The failures would be `status_results_length - filtered_results_length` and that might make the code here less verbose
- but then maybe we want to log the payload IDs that are still pending, in the `Operation not finalized yet` log, wdyt? If so, then I'd just move this `payload_status_results` partitioning logic into its own function to lower verbosity here

Reply:
I adopted the proposal.